src/vulkan/handle_value_with_memory.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright 2018 The Amber Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/vulkan/handle_value_with_memory.h"
+
+namespace amber {
+namespace vulkan {
+namespace {
+
+template <typename T>
+void SetValueForBuffer(void* memory, const std::vector<Value>& values) {
+  T* ptr = static_cast<T*>(memory);
+  for (const auto& v : values) {
+    *ptr = static_cast<T>(v.IsInteger() ? v.AsUint64() : v.AsDouble());
+    ++ptr;
+  }
+}
+
+}  // namespace
+
+HandleValueWithMemory::HandleValueWithMemory() = default;
+
+HandleValueWithMemory::~HandleValueWithMemory() = default;
+
+void HandleValueWithMemory::UpdateMemoryWithData(void* memory,
```

Comment:
It seems somewhat unsafe to have an interface that disables the possibility of checking that you did not overflow the memory buffer.

Reply:
updated.